test/e2e/plain-v0_provisioner_test.go

```diff
@@ -423,8 +423,118 @@ var _ = Describe("plain-v0 provisioner bundleinstance", func() {
+					ProvisionerClassName: plainProvisionerID,
+					BundleName:           bundle.GetName(),
+				},
+			}
+
+			err = c.Create(ctx, biOriginal)
+			Expect(err).To(BeNil(), "failed to create original bundle instance")
+
+			biDuplicate = &rukpakv1alpha1.BundleInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "olm-apis-duplicate",
+				},
+				Spec: rukpakv1alpha1.BundleInstanceSpec{
+					ProvisionerClassName: plainProvisionerID,
+					BundleName:           bundle.GetName(),
+				},
+			}
+
+			err = c.Create(ctx, biDuplicate)
+			Expect(err).To(BeNil(), "failed to create duplicate bundle instance")
+		})
+
+		AfterEach(func() {
+			By("deleting the testing Bundle resource")
+			Eventually(func() error {
+				return client.IgnoreNotFound(c.Delete(ctx, bundle))
+			}).Should(Succeed())
+
+			By("deleting the testing original BundleInstance resource")
+			Eventually(func() error {
+				return client.IgnoreNotFound(c.Delete(ctx, biOriginal))
+			}).Should(Succeed())
+
+			By("deleting the testing duplicate BundleInstance resource")
+			Eventually(func() error {
+				return client.IgnoreNotFound(c.Delete(ctx, biDuplicate))
+			}).Should(Succeed())
+		})
+
+		It("should succeed installation for the original but fail for the duplicate BundleInstance", func() {
+			By("projecting a successful installation status for the original BundleInstance")
+			Eventually(func() bool {
+				if err := c.Get(ctx, client.ObjectKeyFromObject(biOriginal), biOriginal); err != nil {
+					return false
+				}
+				if biOriginal.Status.InstalledBundleName != "" {
+					return false
+				}
+				existing := meta.FindStatusCondition(biOriginal.Status.Conditions, "Installed")
+				return existing != nil
+			})
+
+			By("projecting a failed installation status for the duplicate BundleInstance")
+			Eventually(func() bool {
+				if err := c.Get(ctx, client.ObjectKeyFromObject(biDuplicate), biDuplicate); err != nil {
+					return false
+				}
+				if biDuplicate.Status.InstalledBundleName != "" {
+					return false
+				}
+				existing := meta.FindStatusCondition(biDuplicate.Status.Conditions, "Installed")
+				if existing == nil {
+					return false
+				}
+				// Create what the message should contain
+				looselyExpectedMessage := "rendered manifests contain a resource that already exists"
+				expected := metav1.Condition{
+					Type:    "Installed",
+					Status:  metav1.ConditionStatus(corev1.ConditionFalse),
+					Reason:  "InstallFailed",
+					Message: looselyExpectedMessage,
+				}
+
+				return conditionsLooselyEqual(expected, *existing)
+			}).Should(BeTrue())
+		})
+	})
 })
 
 func conditionsSemanticallyEqual(a, b metav1.Condition) bool {
 	return a.Type == b.Type && a.Status == b.Status && a.Reason == b.Reason && a.Message == b.Message
 }
+
+func conditionsLooselyEqual(a, b metav1.Condition) bool {
+	return a.Type == b.Type && a.Status == b.Status && a.Reason == b.Reason && strings.Contains(b.Message, a.Message)
+}
```

Comment:
Yup - this is exactly why I think 1:1 error message (or in this case status condition message) checking makes tests brittle over time. I think this is fine for now, but we really should move towards a strings.Contains approach imo.

Reply:
So you're thinking instead of having exact equality, all of these should be `.Contains`?